leader-election-impl/src/main/java/com/palantir/leader/health/LeaderElectionHealthCheck.java

```diff
@@ -38,8 +44,29 @@ private double fiveMinuteRate(LeaderElectionServiceMetrics leaderElectionRateFor
         return leaderElectionRateForClient.proposedLeadership().getFiveMinuteRate();
     }
 
-    public LeaderElectionHealthStatus leaderElectionRateHealthStatus() {
-        return getLeaderElectionRateForAllClients() <= MAX_ALLOWED_LAST_5_MINUTE_RATE
-                ? LeaderElectionHealthStatus.HEALTHY : LeaderElectionHealthStatus.UNHEALTHY;
+    private boolean isHealthCheckDeactivated() {
+        if (!healthCheckDeactivated) {
+            healthCheckDeactivated
+                    = Duration.between(timeCreated, Instant.now()).compareTo(HEALTH_CHECK_DEACTIVATION_PERIOD) < 0;
+        }
+        return healthCheckDeactivated;
+    }
+
+    private boolean isHealthy(double leaderElectionRateForAllClients) {
+        return isHealthCheckDeactivated() || (leaderElectionRateForAllClients <= MAX_ALLOWED_LAST_5_MINUTE_RATE);
+    }
+
+    public LeaderElectionHealthReport leaderElectionRateHealthReport() {
+        double leaderElectionRateForAllClients = getLeaderElectionRateForAllClients();
+
+        return isHealthy(leaderElectionRateForAllClients)
+                ? LeaderElectionHealthReport.builder()
+                .status(LeaderElectionHealthStatus.HEALTHY)
+                .leaderElectionRate(leaderElectionRateForAllClients)
+                .build()
+                : LeaderElectionHealthReport.builder()
+                        .status(LeaderElectionHealthStatus.UNHEALTHY)
+                        .leaderElectionRate(leaderElectionRateForAllClients)
+                        .build();
```

Comment:
An interesting way of doing it. Since some of the builder is the same, I suggest you do either:

```java
LeaderElectionHealthStatus status = isHealthy(...) ? LeaderElectionHealthStatus.HEALTHY : LeaderElectionHealthStatus.UNHEALTHY;
return LeaderElectionReport.builder().status(status).leaderElectionRate(...).build();
```

or you could inline `status` to be inside the builder.